frontend/app/(tabs)/lists/index.tsx

```diff
@@ -1,50 +1,80 @@
 /* eslint-disable react/prop-types */
 import { useRouter } from "expo-router";
-import { Button, View, Stack, Card, XStack } from "tamagui";
+import { View, Stack, Card, XStack } from "tamagui";
+import { useForm, Controller } from "react-hook-form";
 import {
   Text,
   FlatList,
   Modal,
   Alert,
   TextInput,
-  SafeAreaView,
+  Button,
+  StyleSheet
 } from "react-native";
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+
 
 export default function One() {
-  const router = useRouter();
   const [modalVisible, setModalVisible] = useState(false);
-  const [name, onChangeNameText] = React.useState("");
-  const [description, onChangeDescriptionText] = React.useState("");
+  const [lists, setLists] = useState([]);
 
-  const lists = [
-    {
-      name: "Breakfast",
-      description: "Some description here...",
-    },
-    {
-      name: "Lunch",
-      description: "Some description here...",
-    },
-    {
-      name: "Dinner",
-      description: "Some description here...",
-    },
-    {
-      name: "Snacks",
-      description: "Some description here...",
+  const {
+    reset,
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      name: "",
+      description: "",
     },
-    {
-      name: "Desserts",
-      description: "Some description here...",
-    },
-  ];
+  })
+ 
+  const onSubmit = async (data) => {
+    try {
+      const response = await fetch('https://open-recipes.onrender.com/recipe-lists', {
+      method: 'POST',
+      headers: {
+          'Content-Type': 'application/json',
+         },
+        body: JSON.stringify(data),
+       });
+
+       if (!response.ok) {
+         throw new Error('Failed to add data');
+       }
+
+      fetchDataFromBackend();
+    } catch (error) {
+      console.error('Error adding data:', error.message);
+    }
+    setModalVisible(!modalVisible)
+  };
+
+  useEffect(() => {
+    fetchDataFromBackend();
+  }, []);
+
+  const fetchDataFromBackend = async () => {
```

Comment:
In order to get a particular users recipe lists you will need to send the jwt auth token in the headers of the request. Abby can show you how to do this. It will be necessary before monday

Reply:
Yeah so eventually the submit end point will need a user ID, to grab that you can use this function: async function getValueFor(key) {
  let result = await SecureStore.getItemAsync(key);
  if (result) {
    alert("🔐 Here's your value 🔐 \n" + result);
  } else {
    alert('No values stored under that key.');
  }
}
use the key "authtoken" 
and then you will need to put that in the header of your fetch statement, it should look something like this: 
headers: {
                Authorization: `Bearer ${authToken}`,
                Accept: "application/json",
              },